src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.cs

```diff
@@ -157,7 +157,7 @@ async Task SyncPlatformModalStackAsync()
 					}
 
 					var page = await PopModalPlatformAsync(animated);
-					page.Parent = null;
+					page.Parent?.RemoveLogicalChild(page);
```

Comment:
Is this the same thing? The code is maybe better but are you setting `Parent = null` still? Would a reverse operation extension be worth something:

```cs
page.RemoveFromLogicalParent();
```

```cs
void RemoveFromLogicalParent(this Element element)
{
  if (element.Parent is Element parent) {
    parent.RemoveLogicalChild(element);
    element.Parent = null;
  }
}
```

Reply:
Or is this already happening because `RemoveLogicalChild` calls `OnChildRemoved`? And that appears to set the parent to null.

Reply:
> Or is this already happening because `RemoveLogicalChild` calls `OnChildRemoved`? And that appears to set the parent to null.

Yes, `RemoveLogicalChild` takes care of setting the parent to null. 

I actually have another branch from this PR where setting the parent literally just routes to `AddLogicalChild` and `RemoveLogicalChild`. I tried doing that with this PR but ran into some issues because `TableView` and `ListView` are setup in a really weird way :-/ 

I have a branch that does this, which I'm hoping to get in for NET8 to just try and standardize everything. But we'll see

![image](https://github.com/dotnet/maui/assets/5375137/1e512423-4032-43b9-adf2-4b6b504a84a9)